addon/utils/selection.js

```diff
@@ -114,6 +116,14 @@ export default {
       }
     }
 
+    // Wipe out duplicates if range selection covered already selected items
+    // e.g item2-3 already selected but shift click item0 through item5 happens
```

Comment:
Visually it works until you count 'selectedItems' growth and you notice duplicates are being sent in by default.